### ISSUE TITLE
nx-serve only works with dev environments

### ISSUE THREAD

**Issue by vespertilian** [studds/nx-aws#77]
Hey, 

So I have an account-based AWS setup (separate accounts for each env). I changed the stack suffix from dev to prod as they are all effectively prod. 

However, I can not serve locally when using layers as we seem to use a `mimic env` var which does not set the layer suffix, layers are also deployed as my-layer-name-[dev/prod].

I was looking at trying to add this however I cannot seem to even build the master branch. There was no `package.lock` file checked in so maybe an update has broken the build?

<img width="1135" alt="Screen Shot 2021-03-30 at 10 09 10 pm" src="https://user-images.githubusercontent.com/441418/112980115-2d441500-91a5-11eb-87eb-8091d076679b.png">
suffixes. 

Can you build the project after an npm install? 

Also, I am thinking maybe mimicEnv useDeployedSuffix ? Or does it mimic other aspects of the build?

**Comment by studds** [comment]
> There was no `package.lock` file checked in so maybe an update has broken the build?

The repo uses yarn now, not npm. So it's got a yarn.lock and you install deps using `yarn`.

That type of error often clears up after removing node_modules and installing again.

I've never tested mimicEnv across accounts so I'm not sure if there's any gotchas there. But essentially all it's doing is finding a stack and grabbing the environment variables from lambdas in that stack, and importing them locally. That's not relevant to layers, because the layer arn is not an environment variable.

The layer arn is derived from a stack output. The relevant parts of the example config:

```
"serve": {
  "builder": "@nx-aws/sam:execute",
  "options": {
    "buildTarget": "api:build",
    "packageTarget": "api:package",
    "importStackOutputs": {
      "TestLayer": {
        "targetName": "layer:deploy",
        "outputName": "TestLayer"
      },
      "LibAsLayer": {
        "targetName": "lib-as-layer:deploy",
        "outputName": "LibAsLayer"
      }
    },
    "mimicEnv": "dev"
  }
},
```

**Comment by vespertilian** [comment]
@studds 



> The repo uses yarn now, not npm. So it's got a yarn.lock and you install deps using `yarn`.

Yes... silly me. I just use NPM in my projects, didn't even think about it. You list it in the readme and everything. Sorry. 


> That's not relevant to layers, because the layer arn is not an environment variable.

The layer arn is  important for the layers as when you load SAM locally it pulls the remote layer from the AWS server. And as the whole stack gets the -dev -prod suffix, and I don't have a dev env (just prod) it cannot pull anything. I have a PR for this. 

I wrote some specs ... I will try to help out with some more in the future.

**Comment by studds** [comment]
@vespertilian thanks so much for the pr, I've been slammed recently and haven't had a chance to look. Do you have some time this week to review together?

**Comment by vespertilian** [comment]
@studds 

Yeah, I think I can do even better but let's talk about it Wednesday or Thursday some time?
